androidApp/src/main/java/com/example/tasky/android/agenda/viewmodel/AgendaViewModel.kt

```diff
@@ -127,6 +132,10 @@ class AgendaViewModel(
                 val newList = screenStateFlow.value.agendas.toMutableList()
                 newList.remove(AgendaItemUi.Item(agendaItem))
                 _screenStateFlow.update { it.copy(agendas = newList.toImmutableList()) }
+
+                alarmRepository.getAgendaAlarm(agendaItem.id)?.let {
+                    alarmScheduler.cancel(it.requestCode)
+                }
```

Comment:
Also, wrap in `applicationScope` to have more of a guarantee that the alarm gets cancelled once the item is deleted.